okhttp/src/test/java/okhttp3/internal/tls/HostnameVerifierTest.java

```diff
@@ -516,6 +516,42 @@
     assertThat(verifier.verify("quux.com", session)).isFalse();
   }
 
+  @Test public void subjectAltNameWithIPAddresses() throws Exception {
+    // $ cat ./cert.cnf
+    // [req]
+    // distinguished_name=distinguished_name
+    // req_extensions=req_extensions
+    // x509_extensions=x509_extensions
+    // [distinguished_name]
+    // [req_extensions]
+    // [x509_extensions]
+    // subjectAltName=IP:0:0:0:0:0:0:0:1,IP:2a03:2880:f003:c07:face:b00c::2,IP:0::5,IP:192.168.1.1
```

Comment:
Wanna do an IPv4 mapped address also?
https://github.com/square/okhttp/issues/4451

Reply:
What do you want the behaviour to be for ::ffff:192.168.1.1?

Reply:
Will land that as a follow up.

Reply:
I think it’s the IPv4 address.

Reply:
What's the reasoning?  Are you self signing certificates but don't want to specify both forms of IPv4 and IPV6 (mapped IPv4) and you expect it to still work for either?

I'm nervous about Postel's law here.  Maybe keeping it explicit is the better option?

Reply:
The canonical form of certain IPv6 addresses (with colons!) is an IPv4 address (no colons!). It’s weird, but it’s how it’s specified and I’ve come to terms with it.

If we’re canonicalizing IP addresses, we should canonicalize all of ’em and test that it works. I expect the test already passes but I’d like coverage of that case.

Reply:
This ok? https://github.com/square/okhttp/pull/5892